CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+
+        public AuthController(IAuthRepository repo, IConfiguration config)
+        {
+            _repo = repo;
+            _config = config;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {
+            //validate request
+            userForRegisterDto.UserName = userForRegisterDto.UserName.ToLower();
+            
+            if (await _repo.UserExists(userForRegisterDto.UserName))
```

Comment:
so instead of mutating the DTO's UserName property, juts pass in the lower case version in this method.

```
await _repo.UserExists(userForRegisterDto.UserName.ToLower()
```

this is totally valid because a string is immutable so you will not be overriding the already existing value.

Reply:
what makes Steam user management cool and fun is you can change your nick name whenever you want.  this is because a user is unique by email address not by username.  this is a design decision of course but user names should be unique to a user.  an email address on the other hand. we can check for uniqueness, use it to send information, then the user can have whatever username they want.

Reply:
Will keep this in mind thank you.